pkg/cmd/init/exec.go

```diff
@@ -353,3 +359,24 @@ func (o *Options) deploySingletonControlplane(kubeClient kubernetes.Interface) e
 	}
 	return nil
 }
+
+func getRegistrationDrivers(o *Options) ([]operatorv1.RegistrationDriverHub, error) {
+	registrationDrivers := []operatorv1.RegistrationDriverHub{}
+	if slices.Contains(o.registrationAuth, "csr") {
+		registrationDrivers = append(registrationDrivers, operatorv1.RegistrationDriverHub{AuthType: "csr"})
+	}
+	if slices.Contains(o.registrationAuth, "awsirsa") {
+		rawConfig, err := o.ClusteradmFlags.KubectlFactory.ToRawKubeConfigLoader().RawConfig()
+		if err != nil {
+			klog.Errorf("unable to load hub cluster kubeconfig: %v", err)
+			return nil, err
+		}
+		hubClusterArn := rawConfig.Contexts[rawConfig.CurrentContext].Cluster
```

Comment:
why the arc is the cluster name? is it alway true on AKS?

Reply:
Yes

Reply:
Adding another option flag called --hub-cluster-arn which can be used in tests as well as in case the hubclusterArn is missing in kubeconfig.